Add support for RedHat

michal800106 opened the pull request:
We want to present support for RedHat by option installing dependencies with yum.

zielonkam committed:
Initial version for RedHat

zielonkam committed:
Revert "Initial version for RedHat"

This reverts commit bf533e8e0086194cd51ebc0ac99e870954057ed3.

zielonkam committed:
Adding yum install openjdk

zielonkam committed:
Merge branch 'adding_yum' into gitlab/master

Commit:
Parameter apache_mirror changed for offline

Commit:
Fix for apache_mirror address

Commit:
Variable apache_mirror changed to kafka_apache_mirror

Commit:
[kafka] Scripts: Network and mutual dependencies

Change-Id: Ifcdf0215e901250eb0b01c990b9898896942c89b

zielonkam committed:
Add firewalld rules

zielonkam committed:
Fix for iptables checking

Commit:
[kafka] Uninstaller polishing

- no longer downloading to /root

Change-Id: I53547d6fcc21241f21bb87c423992b3e15393320

zielonkam committed:
Fix ports for singlehost

Change-Id: Id9cd7ebc536d10f647884c6be36f872450ebe3f8

zielonkam committed:
Add support for RedHat

tkuhlman commented on templates/kafka.service.j2:
```diff
@@ -1,6 +1,7 @@
 [Unit]
 Description=Kafka Daemon
-After=zookeeper.service
```
In some installs zookeeper runs on the same box as kafka. In those cases it is important that zookeeper start first. In cases where zookeeper is not on the same box systemd handles the situation properly and ignores this line. The line should stay for that first case.

tkuhlman commented on defaults/main.yml:
```diff
@@ -1,5 +1,5 @@
 ---
-apache_mirror: http://apache.mirrors.tds.net
+kafka_apache_mirror: http://apache.mirrors.tds.net
```
Why the name change? We have a few things that download from an apache mirror and it is nice to set only one variable to update the mirror for all of our roles.

tkuhlman commented on defaults/main.yml:
```diff
@@ -26,3 +26,5 @@ monasca_log_level: WARN
 
 verify: True
+opt_dir: "/opt"
+download_tmp_dir: "{{ opt_dir }}/.download_dir"
```
This is effectively a rename of kafka_tarball_location I don't see why you want to do that but it is okay if you clean up the previous value. One thing I don't like is the default being a hidden directory, it makes it difficult to find and occasionally it is nice to find the original source.

tkuhlman commented on tasks/install.yml:
```diff
@@ -1,6 +1,19 @@
   apt: name=openjdk-7-jre-headless state=present
+  when: not skip_install and "ansible_os_family" == 'Debian'
+
+- name: Install OpenJDK
```
This looks great. One minor change would be to put Debian or Redhat into the names of the tasks. That way when you see one or the other skip during a run it is immediately clear why.

tkuhlman commented on templates/server.properties.j2:
```diff
@@ -11,9 +11,16 @@ broker.id={{loop.index0}}
 {% endfor %}
 
+{% if adresses_added is not defined %}
+broker.id=0
```
broker.id and port are already set above. What is the purpose of this loop?

michal800106 commented:
Please recheck.

Commit:
Add logging variables to defaults

Change-Id: I2b6c30f731b209eac0c9395b873a6a0ba7a83a6b

michal800106 commented:
I've squashed two commits into one.

tkuhlman commented:
FYI for future note there is no need to bother with squashing. That is a gerrit process that personally I rather do without, it isn't needed at all for github. It won't cause issues in the case so no problems just skip the extra effort next time.

zielonkam committed:
Merge remote-tracking branch 'github_fj/master' into gitlab

Commit:
Access rights for kafka

- setting access rights
- code refactor

Change-Id: I56fefb8b90158af2e9aeb6ae5cc705f6141c1867

witekest commented:
@tkuhlman What is the status of that pull request? Can we merge it?

Commit:
Upgrade kafka

Upgrading kafka version to 0.8.2.2.
For better flexibility added scala_version variable,
becasue kafka in the same version can be download
with different scala version.

Change-Id: Ide676f679f5a991f409a586f7679f9254fecd5d2

Commit:
Cluster integration [kafka]

- added creating user/group
- added status run mode
- removed verify support [launches everything]

Change-Id: Ic4fa7196227aeeb83d22b45ba446d49ab6fc0cc0

Commit:
Fix controller and state-change log severity

Commits add 'monasca_log_level' variable presenting
common log level of entire monasca components
to the state-change and controller appenders.

Prior to that not only exceptions and warnings were
logged but also traces and debugs which resulted
in quickly growing log files.

Change-Id: If867d00669b707bf189c92aca86e9a906c7a81e9
